apps/prairielearn/src/ee/auth/lti13/lti13auth.ts

```diff
@@ -0,0 +1,268 @@
+import { Router, type Request, Response, NextFunction } from 'express';
+import asyncHandler = require('express-async-handler');
+import { Issuer, Strategy, type StrategyVerifyCallbackReq, IdTokenClaims } from 'openid-client';
+import passport = require('passport');
+import { z } from 'zod';
+import { get as _get, some as _some } from 'lodash';
+
+import { loadSqlEquiv, queryAsync } from '@prairielearn/postgres';
+import error = require('@prairielearn/error');
+import * as authnLib from '../../../lib/authn';
+import { selectLti13Instance } from '../../models/lti13Instance';
+import { get as cacheGet, set as cacheSet } from '../../../lib/cache';
+import { getInstitutionAuthenticationProviders } from '../../lib/institution';
+
+const sql = loadSqlEquiv(__filename);
+const router = Router({ mergeParams: true });
+
+const lti13_issuers = {};
+
+// Middleware to cache passport setup
+router.use(
+  asyncHandler(async (req, res, next) => {
+    const lti13_instance = await selectLti13Instance(req.params.lti13_instance_id);
+    const instAuthProviders = await getInstitutionAuthenticationProviders(
+      lti13_instance.institution_id,
+    );
+
+    if (!_some(instAuthProviders, ['name', 'LTI 1.3'])) {
```

Comment:
Nit: let's use `Array.prototype.some` instead of lodash here.

Reply:
Done in ad8bebc